PanModal/Controller/PanModalPresentationController.swift

```diff
@@ -234,12 +246,57 @@ open class PanModalPresentationController: UIPresentationController {
             }
         })
     }
-
+    
+    // MARK: - Actions
+    
+    /**
+     Update long form height and short form height when the keyboard is showing/scrolling and scroll to correct position above the keyboard if keyboard is showing
+     */
+    @objc func keyboardWillShow(notification: Notification) {
+        
+        //if the presented view is already at max height and is scrollable then scroll to above the keyboard when keyboard is shown if needed
+        
+        if let keyboardSize = (notification.userInfo?  [UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
+            
+            if let scrollView = presentable?.panScrollable, pannedToMax {
+                scrollView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: keyboardSize.height, right: 0.0);
+                return
+            }
+            
+            if onShowKeyboardFormYPosition == nil {
+                onShowKeyboardFormYPosition = presentedView.frame.origin.y - keyboardSize.height
+            }
+            
+            guard let onShowKeyboardFormYPosition = onShowKeyboardFormYPosition else { return }
+            shortFormYPosition = onShowKeyboardFormYPosition
+            longFormYPosition = onShowKeyboardFormYPosition
+            snap(toYPosition: onShowKeyboardFormYPosition)
+        }
+    }
+    
+    /**
+     Update long form height and short form height back to original values once keyboard is hidden
+     */
+    @objc func keyboardWillHide(notification: Notification){
+        onShowKeyboardFormYPosition = nil
+        
+        guard let layoutPresentable = presentedViewController as? PanModalPresentable.LayoutType else { return }
+        
+        shortFormYPosition = layoutPresentable.shortFormYPos
+        longFormYPosition = layoutPresentable.longFormYPos
+        
+        transition(to: .shortForm)
+    }
 }
 
 // MARK: - Public Methods
 
 public extension PanModalPresentationController {
+    
+    func observeKeyboard() {
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
```

Comment:
You need to unsubscribe from these.

Reply:
I'm wrong... you don't need to anymore.. I thought that was only the token version.